src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -0,0 +1,299 @@
+                && results[incomingEdgeName.length()][startingEdgeName.length()] == 1 ? -1
+                        : results[incomingEdgeName.length()][startingEdgeName.length()];
+    }
+
+    /**
+     * Check if the parameter characters are different directional characters.
+     *
+     * @param incomingEdgeCharacter
+     *            the incoming Edge's character
+     * @param startingEdgeCharacter
+     *            the starting Edge's character
+     * @return true if the parameter characters are both directional (members of Direction enum) AND
+     *         they are different from one another, false otherwise
+     */
+    private static boolean hasDirectionalCharacterDifference(final char incomingEdgeCharacter,
+            final char startingEdgeCharacter)
+    {
+        return DIRECTIONS.stream().map(Enum::toString).filter(
+                equalCharacter(incomingEdgeCharacter).or(equalCharacter(startingEdgeCharacter)))
+                .count() >= 2;
+    }
+
+    /**
+     * Check if the parameter characters are different numerical characters.
+     *
+     * @param incomingEdgeCharacter
+     *            the incoming Edge's character
+     * @param startingEdgeCharacter
+     *            the starting Edge's character
+     * @return true if both parameter characters are different numbers in [0,9], false otherwise
+     */
+    private static boolean hasNumericalCharacterDifference(final char incomingEdgeCharacter,
+            final char startingEdgeCharacter)
+    {
+        return Character.isDigit(incomingEdgeCharacter) && Character.isDigit(startingEdgeCharacter)
+                && IntStream.range(ZERO, NINE)
+                        .filter(number -> number == (incomingEdgeCharacter - ASCII_OFFSET)
+                                || number == (startingEdgeCharacter - ASCII_OFFSET))
+                        .count() >= 2;
+    }
+
+    /**
+     * Retrieve the minimum value out of parameter numbers.
+     *
+     * @param numbers
+     *            the numbers on which to operate
+     * @return the minimum of those numbers
+     */
+    private static int min(final int... numbers)
+    {
+        return Arrays.stream(numbers).min().orElse(Integer.MAX_VALUE);
+    }
+
+    /**
+     * Wrapper for getLevenshteinDistance(). Handles cases where the incomingEdge doesn't have a
+     * name and where the incomingEdge has the same name as the startingEdge before computing the
+     * Levenshtein distance.
+     *
+     * @param incomingEdge
+     *            the next edge in the search area
+     * @param startEdge
+     *            the edge from which the search started
+     * @return the Levenshtein distance between two edge's names, or -1 if the incoming edge doesn't
+     *         have a name, or 0 if the names are the same
+     */
+    @SuppressWarnings("squid:S3655")
+    private static int similarityIndex(final Edge incomingEdge, final Edge startEdge)
```

Comment:
Since this gives the similarity index of two strings, I think we can just pass the name tags instead of the edge itself.